test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -141,6 +141,38 @@
         ;; now the pool is flushed, so the constants should be cleared
         (is (true? (verify-no-constants pool-context 4)))))))
 
+(deftest ^:integration admin-api-multiple-flush-jruby-pool-test
```

Comment:
This test is kinda weak, but I couldn't think of a good system-level way to intercept the flushing process midstream and insert another request while being able to predict whether or not an instance from the new pool would be created before a second flush would come along and wipe it out first.  So this test doesn't ensure that the flushes aren't done synchronously - i.e., that the flush isn't completed before each HTTP response to the flush request is delivered - and doesn't determine the state of each of the pool instances after the first or second flush (only after the last one).  It could, however, expose any problems that might arise from the pool not being updated atomically like it should and that all of the flush requests are at least processed.